src/arrayMethodSort.js

```diff
@@ -4,8 +4,53 @@
  * Implement method Sort
  */
 function applyCustomSort() {
-  [].__proto__.sort2 = function(compareFunction) {
-    // write code here
+  function compareDefault(a, b) {
+    const aString = String(a);
+    const bString = String(b);
+    const areInSameCase =
+      (isUpperCase(aString) && isUpperCase(bString)) ||
+      (!isUpperCase(aString) && !isUpperCase(bString));
+
+    if (areInSameCase) {
+      return aString.localeCompare(b);
+    }
+
+    if (isUpperCase(a)) {
+      return -1;
+    } else {
+      return 1;
+    }
+  }
+
+  function isUpperCase(item) {
+    return item.toLowerCase() !== item;
+  }
+
+  [].__proto__.sort2 = function (compareFunction = compareDefault) {
+    let itemToBePlacedAtTheEnd = this[0];
+    let currentEnd = this.length;
+    let currentSearchIndex = 0;
+    let itemIndex = 0;
+
+    for (let repeatCount = 1; repeatCount < this.length; repeatCount++) {
+      while (currentSearchIndex < currentEnd) {
+        if (
+          compareFunction(this[currentSearchIndex], itemToBePlacedAtTheEnd) > 0
+        ) {
+          itemToBePlacedAtTheEnd = this[currentSearchIndex];
+          itemIndex = currentSearchIndex;
+        }
+        currentSearchIndex++;
+      }
+      this[itemIndex] = this[currentEnd - 1];
+      this[currentEnd - 1] = itemToBePlacedAtTheEnd;
+      currentEnd--;
+      currentSearchIndex = 0;
+      itemToBePlacedAtTheEnd = this[0];
+      itemIndex = 0;
+    }
+
+    return this;
   };
```

Comment:
```suggestion
  [].__proto__.sort2 = function (compareFunction = compareDefault) {
    for (let i = 0; i < this.length - 1; i++) {
      for (let j = i + 1; j < this.length; j++) {
        if (compareFunction(this[i], this[j]) > 0) {
          const temp = this[i];

          this[i] = this[j];
          this[j] = temp;
        }
      }
    }
```

Your solution looks too complicated and we can simplify it. Due to sort method documentation we need to swap elements if the callback function returns a value greater than zero